tools/codegen/pkg/schemapatch/generator.go

```diff
@@ -110,10 +112,14 @@ func (g *generator) GenGroup(groupCtx generation.APIGroupContext) error {
 		klog.V(1).Infof("%s API schema for for %s/%s", action, groupCtx.Name, version.Name)
 
 		if err := g.genGroupVersion(groupCtx.Name, version, versionPaths); err != nil {
-			return fmt.Errorf("could not run schemapatch generator for group/version %s/%s: %w", groupCtx.Name, version.Name, err)
+			errs = append(errs, fmt.Errorf("could not run schemapatch generator for group/version %s/%s: %w", groupCtx.Name, version.Name, err))
```

Comment:
Or `\n` at the line start, might looks better in output.

Reply:
In this case, this would change the error in a lot of other cases as well, so for now, I think I'm going to leave as is. I think for the most part people will just rerun the whole generator so I expect most people won't be too fussed about this

Reply:
The alternative is to extend the Diff util to pass in the file name 🤔

Reply:
>The alternative is to extend the Diff util to pass in the file name 🤔

A variant, but without it, the code was quite cool copy-paste ready pretty-printer.

I suggest to make it as it was and go with it.

>I think for the most part people will just rerun the whole generator so I expect most people won't be too fussed about this

I agree.

Reply:
Could always change the interface to make it have options you can pass in for the diff, eg, passing in the file name and make that optional in the header 🤔 Maybe in a future iteration